external/llvm-project/mlir/lib/Conversion/TosaToLinalg/TosaToLinalgNamed.cpp

```diff
@@ -318,10 +371,16 @@ class ConvConverter : public OpConversionPattern<TosaConvOp> {
 
     Value conv = rewriter
                      .create<LinalgConvOp>(
-                         loc, resultTy, ValueRange{input, weight},
+                         loc, newResultTy, ValueRange{input, weight},
                          ValueRange{zeroTensor}, strideAttr, dilationAttr)
                      ->getResult(0);
 
+    if (isConv2DOp && group > 1) {
```

Comment:
nit : we just do a very early check on isConv2DOp and fail ; so that we dont have to keep checking whether its a conv2d